src/components/views/rooms/EventTile.js

```diff
@@ -484,12 +484,14 @@ module.exports = withMatrixClient(React.createClass({
         // Info messages are basically information about commands processed on a room
         const isInfoMessage = (eventType !== 'm.room.message' && eventType !== 'm.sticker');
 
-        const EventTileType = sdk.getComponent(getHandlerTile(this.props.mxEvent));
+        const tileHandler = getHandlerTile(this.props.mxEvent);
         // This shouldn't happen: the caller should check we support this type
         // before trying to instantiate us
-        if (!EventTileType) {
-            throw new Error("Event type not supported");
+        if (!tileHandler) {
+            const {mxEvent} = this.props;
+            throw new Error(`Event type not supported: type:${mxEvent.getType()} isState:${mxEvent.isState()}`);
```

Comment:
This isn't exactly a fix; it just throws a different error. React will become wibbly because we're throwing an error in `render`.

How about we `console.warn` loudly and display a placeholder in the style of a notice that reads "This event could not be displayed"? (If we just returned `null`, the timeline might look odd, considering the `continuation` prop.)

(If we were on React16 we'd use error boundaries.)